File: saif_akib/1480/runningSum.js
//TC: O(n)
//MC: O(n)
var runningSum = function(nums) {
    const output = new Array(nums.length);;
    let sum = 0;
    nums.forEach((num, i) => {
        sum += nums[i];
        output[i] = sum;
    });
    return output;
};
const arr = [1, 2, 3, 4];
console.log(runningSum(arr));

File: rafsun_jani/aug25/1480/runningSum.java
class Solution {
    public static int[] runningSum(int[] nums) {
        int[] result = new int[nums.length];
        int sum = 0;
        for (int i = 0; i < nums.length; i++) {
            sum += nums[i];
            result[i] = sum;
        }
        return result;
    }

    public static void main(String[] args) {
        int[] numbers = {1, 3, 4, 5};
        System.out.print(Arrays.toString(runningSum(numbers)));
    }
}